### PR TITLE
Reduce tolerance of bad Cassandra hosts

### PR DESCRIPTION
**Pull request by Jolyon-S**
**Goals (and why)**:
We've seen multiple instances internally where we are able to establish a connection to Cassandra, yet then fail to actually use the connection. I have a very strong suspicion that we are able to create the socket (and thus the regular `socketTimeout` is not relevant), but that we then do not read anything from the connection. In this case, the `socketQueryTimeout` is the relevant one, but this is set at 62 seconds (where the other is 2 seconds). This means that we fail very slowly here, and with the retries on a host, this can cause us to wait for up to 126 seconds while attempting to talk to a bad host.

The aim here is to fail faster by reducing this timeout for an initial call to Cassandra when creating a connection.

**Implementation Description (bullets)**:
* Reduce the initial `socketQueryTimeout` to a configured value, defaulting to 5 seconds
* Set this value back to the previous timeout after performing the `login` call to Cassandra

**Testing (What was existing testing like?  What have you done to improve it?)**:
NONE - the testing around this codepath is admittedly sparse. I'm open to suggestions here. I will do at least some internal validation that this doesn't cause things to blow up (and also should give us an idea of how long login takes thanks to the added telemetry), but I appreciate that this may not be sufficient.

**Concerns (what feedback would you like?)**:
* Testing as noted above
* Configuring what a sensible initial timeout is - I'm not sure what a sensible timeout for this call would be in a good state. I suspect 5 seconds is generally generous, but then there is always a fairly wide deviation in Cassandra performance from one install to the next.

**Where should we start reviewing?**:
`CassandraClientFactory`.

**Priority (whenever / two weeks / yesterday)**:
ASAP.

### CODE REVIEW

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraKeyValueServiceConfig.java [hunk 1]
```diff
@@ -325,6 +325,21 @@ default int socketQueryTimeoutMillis() {
         return 62 * 1000;
     }
 
+    /**
+     * When a Cassandra node is down or acting malignantly, it is plausible that we succeed in creating a socket but
+     * subsequently do not read anything, and thus are at the mercy of the {@link #socketQueryTimeoutMillis()}. This is
+     * particularly problematic on the creation of transaction managers and client pools in general: we can end up in
+     * a state where a query to a bad host blocks us for up to 126 seconds with default settings (due to retrying three
```
Comment (jeremyk-91) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraKeyValueServiceConfig.java:332]: Should this be 186?

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientFactory.java [hunk 2]
```diff
@@ -96,19 +97,17 @@ private CassandraClient instrumentClient(Client rawClient) {
         return client;
     }
 
-    private Cassandra.Client getRawClientWithKeyspace(
-            InetSocketAddress inetSocketAddress, CassandraKeyValueServiceConfig kvsConfig) throws TException {
-        Client ret = getRawClient(inetSocketAddress, kvsConfig, sslSocketFactory);
+    private Cassandra.Client getRawClientWithKeyspace() throws TException {
```
Comment (jeremyk-91) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientFactory.java:100]: after your refactor, the name of this method doesn't really make sense. maybe `getRawClientWithKeyspaceSet`?

**Review by jeremyk-91 — APPROVED** [review]
👍

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraKeyValueServiceConfig.java (hunk 1 above)
Comment (Jolyon-S) [reply atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraKeyValueServiceConfig.java:332]: duh

nice catch

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientFactory.java (hunk 2 above)
Comment (Jolyon-S) [reply atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientFactory.java:100]: that's reasonable, sure

**Pull request merged**